### PR TITLE
feat: use en as DB locale default [DHIS2-17408]

### PR DESCRIPTION
**Pull request by jbee**
Similar to UI locale the DB uses a locale default (EN)

### CODE REVIEW

**Review by david-mackessy — APPROVED** [review]

**Comment by jbee** [comment]
Keeps failing `PdfFormControllerTest.testGetDataSetPdfForm:62` now - so must be related

**Comment by vietnguyen** [comment]
I think the test `testGetDataSetPdfForm` can be removed because it was added to support the case when DB_Locale is null.
Now with this fix it will never be null hence the test is redundant.

File: dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SettingKey.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
  */
 public enum SettingKey {
   UI_LOCALE("keyUiLocale", LocaleManager.DEFAULT_LOCALE, Locale.class),
-  DB_LOCALE("keyDbLocale", Locale.class),
+  DB_LOCALE("keyDbLocale", LocaleManager.DEFAULT_LOCALE, Locale.class),
```
Comment (vietnguyen) [inline dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SettingKey.java:62]: This fix does not satisfy a conner case:
- An instance has DB_Locale = null. User enter value in Chinese.  The expected behavior is the raw value ( in Chinese ) should be returned. However with this fix, the DB_Locale is English so the return value will be the translated value in English, which is taken from `translations` jsonb column. 
- If the English translation value does not exist then the raw value will be returned.

Please have a look at [this code](https://github.com/dhis2/dhis2-core/blob/b2f0b6375c8c7ce21264c4300015e4bca0292939/dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseIdentifiableObject.java#L336)

**Comment by jbee** [comment]
Closing in favor of new PR that uses UI locale in the job as a fallback

**Pull request closed without merging**